test/parallel/test-internal-net-isLoopback.js

```diff
@@ -0,0 +1,32 @@
+// Flags: --expose-internals
+'use strict';
+require('../common');
+const assert = require('assert');
+const net = require('internal/net');
+
+const loopback = [
+  'localhost',
+  '127.0.0.1',
+  '127.0.0.255',
+  '127.1.2.3',
+  '[::1]',
+  '[0:0:0:0:0:0:0:1]',
+];
+
+const loopbackNot = [
+  'example.com',
+  '192.168.1.1',
+  '10.0.0.1',
```

Comment:
Should we really warn for these? IMHO binding to IP in private subnet is usually fine.

Reply:
The condition when to warn was discussed in https://github.com/nodejs/node/pull/23756#pullrequestreview-166769614 and https://github.com/nodejs/node/pull/23756#issuecomment-431823915. So, I think we should warn for all non-loopback addresses. WDYT?